colossalai/booster/plugin/gemini_plugin.py

```diff
@@ -403,11 +405,13 @@ def __init__(
         self.verbose = verbose
 
         self.tp_size = tp_size if self.enable_tensor_parallelism else 1
-        self.dp_size = dist.get_world_size() // self.tp_size
-        assert self.dp_size > 1, f"The size of the DP group should be greater than 1. Please reduce the TP group size."
-        self.pg_mesh = ProcessGroupMesh(self.dp_size, self.tp_size)
-        self.dp_group = self.pg_mesh.get_group_along_axis(DP_AXIS)
+        self.ddp_size = ddp_size
+        self.use_ddp = self.ddp_size > 1
+        self.zero_size = dist.get_world_size() // (self.tp_size * self.ddp_size)
+        self.pg_mesh = ProcessGroupMesh(self.zero_size, self.tp_size, self.ddp_size)
+        self.zero_group = self.pg_mesh.get_group_along_axis(ZERO_AXIS)
         self.tp_group = self.pg_mesh.get_group_along_axis(TP_AXIS)
+        self.ddp_group = self.pg_mesh.get_group_along_axis(DDP_AXIS)
```

Comment:
You'd better not create new process group if extra-dp size is `1`, as creating process group costs a lot of time. Same for tp